src/cpu/x64/jit_uni_reorder.cpp

```diff
@@ -179,7 +179,7 @@ struct jit_uni_reorder_kernel_f32_t : public kernel_t, public jit_generator {
                 && IMPLICATION(utils::one_of(bf16, p.itype, p.otype),
                         mayiuse(avx512_core) || mayiuse(avx2_vnni_2))
                 && IMPLICATION(utils::one_of(f16, p.itype, p.otype),
-                        mayiuse(avx512_core_fp16) || mayiuse(avx2_vnni_2))
+                        mayiuse(avx512_core_fp16) || mayiuse(avx2))
```

Comment:
What's the use case for fp16 on avx2?

Reply:
This change is needed to restore load time in case of fp16 models. I moved CF in bounds of https://github.com/openvinotoolkit/openvino/pull/18079 to the plug-in side

Reply:
@dzarukin maybe do you knowif this patch can break something? I've tested locally on CPU wtthout `avx512_core_fp16` support and haven't got any errors

Reply:
For pure oneDNN I wouldn't expect much issues until avx2 starts supplying fp16. It's more about documented restraints, and when they are moved to some "nonsense", it brings a lot of question. Given that there are absolutely no comments neither in the code nor in commit, someone who will try to work towards fork elimination will spend 100x more time trying to understand why this change was done.

Reply:
@dzarukin Thanks. I agree we need properly explain the change. 

@antonvor please modify commit message:
1. Its name should be aligned with our naming convention (see [other commits](https://github.com/openvinotoolkit/oneDNN/commits/v3.2_for_ie_master?before=33bb2b261d3829162395aaa9bbe8c1c5b139e855+35&branch=v3.2_for_ie_master&qualified_name=refs%2Fheads%2Fv3.2_for_ie_master))
2. Update commit message with explanation why isa relaxation was made.

Reply:
I updated commit message and add comments, @dmitry-gorokhov and @dzarukin take a look please

Reply:
Thanks, resolving.

Reply:
@antonvor Would still ask you to correct commit message since it is not WA and motivation is not fullt correct. Proposal:

```
[FORK][FEATURE] jit_uni_reorder: relaxed isa condition to enable FP16 precision support on AVX2 platforms

The feature is required to enable fast FP16->BF16/FP32 precision conversion. Most commonly used for fast weights decompression which improves model compilation time.
```

Reply:
done